.pipelines/templates/run-upstream-e2e.yaml

```diff
@@ -0,0 +1,12 @@
+steps: 
+  - bash: |
+      git clone -b ${UPSTREAM_BRANCH} $(upstream.repo) $(repo.path)/osm
+    displayName: Checkout $(upstream.repo) @ ${UPSTREAM_BRANCH}
+  - bash: |
+      cd osm
```

Comment:
it seems like you are cloning to `osm` then changing directory to `osm/osm`. can you just clone to repo.path then use osm?

Reply:
Yes, I'm still not clear on the need for two separate clones checked out to different branches

Reply:
`fatal: destination path '/home/azureuser/agent/_work/17/s' already exists and is not an empty directory.` 

The pipeline gives this error when I try to clone into $(repo.path)

Reply:
you need to change `repo.path` to be `$(System.DefaultWorkingDirectory)/osm` https://github.com/Azure/osm-azure/blob/8c287229398fc280e0feae9298f36809b6ebe2e5/.pipelines/e2e-job.yaml#L16

Reply:
resolved by just removing $(repo.path) and using $(System.DefaultWorkingDirectory). However, I still need to use $(System.DefaultWorkingDirectory)/osm when cloning the upstream branch.